moveit_core/collision_detection/include/moveit/collision_detection/collision_log.h

```diff
@@ -0,0 +1,50 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2019, Acutronic Robotics AG
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Willow Garage nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Víctor Mayoral Vilches */
+
+#ifndef MOVEIT_COLLISION_DETECTION_COLLISION_LOG_
+#define MOVEIT_COLLISION_DETECTION_COLLISION_LOG_
+
+#include "rclcpp/rclcpp.hpp"
+
+namespace collision_detection
+{
+// Logger
+rclcpp::Logger logger_collision_detection = rclcpp::get_logger("collision_detection");
```

Comment:
@davetcoleman, let me know if you'd prefer this in the cpp files instead, I recall having discussed that. Note that it'll require the same definition in different CPP files thereby AFAIK we should either a) keep this method, b) make the same definition in different CPP files or c) make _different_ definitions in different CPP files (e.g. collision_detection.file1, collision_detection.file2, etc)

Reply:
I think that putting the logger in a separate file is overkill and will result in a lot of extra files. Each header file or class can have it's own Logger that reflects either the source file's name or the class name. (eg. `"collision_robot_allvalid"`. This makes it easier for the user to track down the source of warnings, info etc... It also enables them to selectively filter better. This sounds a lot like the third option that you presented.

Reply:
I think we would prefer to not add so many files. It would be better for each cpp to have it's own logger. We may even choose to have multiple loggers in a CPP when there are multiple classes.

Reply:
https://github.com/ros-planning/moveit2/pull/40/commits/97e461f7ec5865ba3e12bd4c5269ebb00fa53b73